### PR TITLE
Import karmada in go mod and update vendor

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Import karmada in go mod and update vendor.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -77,51 +78,51 @@ require (
...
+	k8s.io/client-go v0.27.3
 	k8s.io/cloud-provider v0.0.0
-	k8s.io/cluster-bootstrap v0.0.0
-	k8s.io/code-generator v0.0.0
-	k8s.io/component-base v0.0.0
-	k8s.io/component-helpers v0.0.0
+	k8s.io/cluster-bootstrap v0.27.3
+	k8s.io/code-generator v0.27.3
+	k8s.io/component-base v0.27.3
+	k8s.io/component-helpers v0.27.3
```
Comment (RainbowMango) [inline go.mod:108]: Can you remind me why need to change this?

Comment (XiShanYongYe-Chang) [reply go.mod:108]: When executing the command `run hack/pin-dependency.sh && hack/update-vendor.sh`, it will auto-modify it.
Do I explain right @wm775825

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Comment by RainbowMango** [comment]
I'm trying to figure out the reason why the robot response `Merge type "merge" disallowed by repo settings`.

**Comment by RainbowMango** [comment]
/lgtm
/approve

Just configured branch protection rules as per karmada repo.

**Review by kevin-wangzefeng — COMMENTED** [review]
I don't think it's a good idea to import karmada-io/karmada to this repo.
The `karmada-io/kubernetes` is created for developing a fleet-apiserver-machinery. That means eventually, the repo `karmada-io/karmada` would import `karmada-io/kubernetes`, so that we can maintain fleet-apiserver code there.

Even in an alpha implementation, if it's unavoidable to depending on Karmada API definitions, e.g. karmada clusters, you may import `karmada-io/api` instead.

**Comment by RainbowMango** [comment]
> Even in an alpha implementation, if it's unavoidable to depending on Karmada API definitions, e.g. karmada clusters, you may import karmada-io/api instead.

Agree.  In addition to `karmada-io/api` we still need another repo to hold the Karmada clients. So, in order not to block the feature, let's move this forward. And, I'll open an issue to track the new repo thing. After that, the dependencies of `karmada-io/karmada` will be removed then.

**Pull request merged**